pdl_syntax.md

```diff
@@ -0,0 +1,828 @@
+    enum anotherEnum {
+
+      /**
+       * description about FOOFOO
+       */
+      FOOFOO
+
+      /**
+       * description about BARBAR
+       */
+      BARBAR
+    }]
+}
+```
+
+## Record Type
+Records contain any number of fields, which can be primitive type, enums, unions, maps and arrays.
+
+A basic record type can contain a few fields.
+
+For example:
+```
+import org.example.time.DateTime
+
+record Example {
+  field1: string
+  field2: optional int
+  field3: DateTime
+}
+```
+Record fields can be optional.
+
+For example :
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * A foo record
+ */
+record Foo {
+   field1: string
+   field2: optional string
+}
+```
+
+Record fields may have default values.
+
+For example :
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * A foo record
+ */
+record Foo {
+   field1: string
+   field2: string = "message"
+}
+```
+
+A optional field may have default value.
+
+For example : 
+```
+namespace com.linkedin.pegasus.generator.test.idl.records
+
+record WithOptionalPrimitiveDefault {
+  intWithDefault: optional int = 1
+}
+```
+
+### Doc Strings
+
+Types and fields may be documented using “doc strings”.
+
+For example : 
+```
+/**
+ * Doc strings may be added to types. This doc should describe the purposes
+ * of the Example type.
+ */
+record Example {
+  /**
+   * Doc strings may also be added to fields.
+   */
+   field1: string
+
+   /** Doc strings can be single line.*/
+   field2: int
+}
+```
+
+**Note:**
+If you use Java comment style for doc string, e.g "// Doc String", those doc strings will not be stored in in-memory schema.
+
+
+### Import
+
+1. If the type is outside the root namespace of the document, you need to add it as an import.
+
+2. If the type is declared outside the document, you need to add it as an import.
+
+For example :
+```
+namespace com.linkedin.pegasus.generator.test.idl.imports
+
+import com.linkedin.pegasus.generator.test.idl.records.Simple
+
+record Example {
+  /**
+   * Requires an import since this type is outside the root namespace and is not declared in this file.
+   */
+  externalOutsideNS: Simple
+}
+```
+
+**Note:** 
+- When multiple referenced types with the same unqualified name may be imported, 
+the type with the alphabetically first namespace is chosen. 
+(e.g. "com.a.b.c.Foo" is chosen over "com.x.y.z.Foo") 
+
+- Any type that is not imported and is not within the namespace from which it's referenced must be referenced by
+fully qualified name.
+
+
+### Deprecation
+Types and fields can be deprecated by adding @deprecated annotation.
+
+For example :
+
+```
+@deprecated = "Use record X instead."
+record Example {
+  @deprecated = "Use field x instead."
+  field: string
+}
+```
+
+### Including fields
+
+Record can include fields from other record.
+
+For example : 
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * Bar includes fields of Foo, Bar will have fields f1 from itself and b1 from Bar
```

Comment:
Comment is wrong, f1 and b1 are switched

Reply:
Fixed